dotnet/src/SemanticKernel/Configuration/KernelConfig.cs

```diff
@@ -201,7 +201,7 @@ public bool HasEmbeddingsBackend(string label, Func<BackendConfig, bool>? condit
     /// <returns>The updated kernel configuration.</returns>
     public KernelConfig SetRetryMechanism(IRetryMechanism? retryMechanism = null)
     {
-        this._retryMechanism = retryMechanism ?? new PassThroughWithoutRetry();
+        this._retryMechanism = retryMechanism ?? new RetryWithDelay();
```

Comment:
There are several things that can be retried, but what we were looking for was a strategy for HTTP requests. The current situation is a bit ambiguous, looking at the implementation and at the names.

I would like to rename field, class and interface to convey that:
* _httpRetryStrategy
* DefaultHttpRetryStrategy
* IHttpRetryStrategy
* make sure the code retries only for HTTP exceptions

wdyt?

Reply:
Well, I'm about to add a handful of changes based on the list of asks in your task, i definitely understand there is some ambiguity. Right now though, it'll retry on AIExceptions -- by default transient ones -- RequestTimeout, ServiceError, and Throttling.

Let's have a conversation next week about it.

Reply:
I think it makes sense to change things such that this only applies to the HTTP layer, but I'd request that be a separate change, that includes per function configuration, and to continue with this current improvement that gives a decent out of the box solution for the issue developers run into.

Reply:
Nevermind, I will make this all HTTP specific and move logic into OpenAIClientAbstract for compare/contrast.

Reply:
Okay, big set of changes added.

Reply:
I think I've resolved these concerns, will do another pass on naming (holler if I missed any). Thanks!